### PR TITLE
Feature/marine obs yaml files

### PR DESCRIPTION
**Pull request by AndrewEichmann-NOAA**
Adds the observation source yaml files from godas (godas/soca/obs) to soca test.

Notes: All are tested with Guillaume's tutorial rig and at least don't prevent the analysis from being created. Certain cases in effect duplicate existing files (icefb_GDR.yaml for the existing icefb_gdr.yaml, temp_profile_fnmoc_godas.yaml
for temp_profile_fnmoc.yaml, and salt_profile_fnmoc_godas.yaml
salt_profile_fnmoc.yaml) but were included as they have QC filtering. The letkf sections in all the yamls have been removed. Certain keywords/sections identified as unrecognized aliases were removed:

```
  - *obs_land_mask
```

under

`  obs filters:
`

and 

```
    distribution:
      name: *obs_distribution
```

Testing of sss_smos_esa failed as the db file /scratch2/NCEPDEV/marineda/r2d2/gdas_marine/ob/s2s_v1/P1D/2021-03-23/gdas_marine.s2s_v1.ob.P1D.sss_smos_esa.2021-03-23T00:00:00Z.nc4 had bad permissions.

Files added:
parm/soca/obs/config/adt_3a_egm2008.yaml
parm/soca/obs/config/adt_3b_egm2008.yaml
parm/soca/obs/config/adt_6a_egm2008.yaml
parm/soca/obs/config/adt_c2_egm2008.yaml
parm/soca/obs/config/adt_coperl4.yaml
parm/soca/obs/config/adt_j2_egm2008.yaml
parm/soca/obs/config/adt_j3_egm2008.yaml
parm/soca/obs/config/adt_sa_egm2008.yaml
parm/soca/obs/config/icec_nsidc_nh.yaml
parm/soca/obs/config/icec_nsidc_sh.yaml
parm/soca/obs/config/icec_ssmis_f17_north.yaml
parm/soca/obs/config/icec_ssmis_f17_south.yaml
parm/soca/obs/config/icec_ssmis_f18_north.yaml
parm/soca/obs/config/icec_ssmis_f18_south.yaml
parm/soca/obs/config/icefb_GDR.yaml
parm/soca/obs/config/oc_aqua.yaml
parm/soca/obs/config/oc_dineof3.yaml
parm/soca/obs/config/oc_noaa20.yaml
parm/soca/obs/config/oc_snpp.yaml
parm/soca/obs/config/salt_profile_fnmoc_godas.yaml
parm/soca/obs/config/salt_profile_wod.yaml
parm/soca/obs/config/sss_smap_jpl.yaml
parm/soca/obs/config/sss_smos.yaml
parm/soca/obs/config/sss_smos_esa.yaml
parm/soca/obs/config/sss_trak_fnmoc.yaml
parm/soca/obs/config/sst_amsr2_l3u.yaml
parm/soca/obs/config/sst_drifter.yaml
parm/soca/obs/config/sst_gmi_l3u.yaml
parm/soca/obs/config/sst_metopa_l3u_so025.yaml
parm/soca/obs/config/sst_metopb_l3u_so025.yaml
parm/soca/obs/config/sst_metopc_l3u_so025.yaml
parm/soca/obs/config/sst_noaa18_l3u_so025.yaml
parm/soca/obs/config/sst_noaa19_l3u_so025.yaml
parm/soca/obs/config/sst_ostia.yaml
parm/soca/obs/config/sst_ship_fnmoc.yaml
parm/soca/obs/config/sst_trak_fnmoc.yaml
parm/soca/obs/config/sst_viirs_n20_l3u_so025.yaml
parm/soca/obs/config/sst_viirs_npp_l3u_so025.yaml
parm/soca/obs/config/sst_windsat_l3u.yaml
parm/soca/obs/config/temp_profile_fnmoc_godas.yaml

### CODE REVIEW

**Comment by guillaumevernieres** [comment]
:+1: @AndrewEichmann-NOAA . I'll take a closer look tomorrow, but we definitely need to add the land filter, and set the io pool to 1. 
I think it has to be done to all the yaml files unfortunately.

```yaml
observations:
- obs space:
    obsdataout:
      engine:
        type: H5File
        obsfile: .....
    io pool:
      max pool size: 1
...
```
unless we can do something like this in the `var.yaml`:
```yaml
  observations:
    io pool:
      max pool size: 1
    observations:  $<< ${OBS_YAML}
```
But I don't think so, worth a try ... Any idea @CoryMartin-NOAA ? @RussTreadon-NOAA ?

The land mask filter to be added is this one:
```yaml
  filter: Domain Check
  where:
  - variable: {name: sea_area_fraction@GeoVaLs}
    minvalue: 0.9
```

**Comment by AndrewEichmann-NOAA** [comment]
@guillaumevernieres Regarding the land mask (and maybe the distribution entry), would it be possible to implement the same alias as in godas? I know nothing about it, so I assume it's easy. ;-)

**Comment by guillaumevernieres** [comment]
> @guillaumevernieres Regarding the land mask (and maybe the distribution entry), would it be possible to implement the same alias as in godas? I know nothing about it, so I assume it's easy. ;-)

Yes, it's easy enough, but I really don't like these yaml anchors, specially if it's just to replace 3-4 lines.

**Comment by CoryMartin-NOAA** [comment]
I would just put the distribution (i.e. io pool) thing in every file. It's relatively few places that this has to be done and once it's there, it will be untouched (hopefully)

**Comment by ShastriPaturi** [comment]
@AndrewEichmann-NOAA I followed with what you were doing so no additional comments. I will approve once you have added the necessary keys.

**Review by guillaumevernieres — DISMISSED** [review]
Looks good.

**Comment by AndrewEichmann-NOAA** [comment]
@guillaumevernieres That was just a preliminary - many more coming!

**Comment by guillaumevernieres** [comment]
> @guillaumevernieres That was just a preliminary - many more coming!

:) I clearly didn't read beyond the first file @AndrewEichmann-NOAA !

**Comment by AndrewEichmann-NOAA** [comment]
The icec_ssmis sources are failing at the analysis step with the message `cannot find field cicen` in `analysis/var.out`

**Comment by guillaumevernieres** [comment]
> The icec_ssmis sources are failing at the analysis step with the message `cannot find field cicen` in `analysis/var.out`

Add the seaice obs yaml file but ignore the error msg. I'm adding sea ice DA as I write this ... But thanks for making sure I had not done that work yet @AndrewEichmann-NOAA !

**Comment by AndrewEichmann-NOAA** [comment]
oc_aqua, oc_noaa20, and oc_snpp  end with the message:
` sea_surface_chlorophyll is specified as a simulated variable but it has not been specified as an observed or a derived variable. caught in  (/scratch1/NCEPDEV/da/Andrew.Eichmann/gdasapp/GDASApp/oops/src/oops/runs/Run.cc +192 execute)
`

**Review by ShastriPaturi — DISMISSED** [review]
Thanks @AndrewEichmann-NOAA

**Comment by guillaumevernieres** [comment]
> oc_aqua, oc_noaa20, and oc_snpp end with the message: `sea_surface_chlorophyll is specified as a simulated variable but it has not been specified as an observed or a derived variable. caught in (/scratch1/NCEPDEV/da/Andrew.Eichmann/gdasapp/GDASApp/oops/src/oops/runs/Run.cc +192 execute)`

same as the sea ice case @AndrewEichmann-NOAA . Add the yaml and ignore the error msg ...

**Comment by AndrewEichmann-NOAA** [comment]
I guess I didn't catch it last time, but `salt_profile_fnmoc_godas.yaml` (previously `salt_profile_fnmoc.yaml`, but renamed due to the existing yaml of the same name) seems contradictory about where it's getting its data

**Comment by AndrewEichmann-NOAA** [comment]
ok, I think that's it

**Comment by ShastriPaturi** [comment]
Only one comment:
@AndrewEichmann-NOAA can you check if the the obsdata for icefb is icefb_GDR_YYYYMMDD.nc4 in the obsdb?
All the current data are icefb_gdr_YYYYMMDD.nc4 and the keys "GDR" should be in lowercase.
I think renaming all GDR to gdr will be helpful.
@guillaumevernieres the earlier icefb_GDR_*nc were only having one swath, if I am not wrong.

**Comment by AndrewEichmann-NOAA** [comment]
@ShastriPaturi Which obsdb? There was an already existing icefb_gdr.yaml GDASApp. The icefb_GDR.yaml was transferred from godas and is similar aside from the QC.

**Comment by ShastriPaturi** [comment]
> @ShastriPaturi Which obsdb? There was an already existing icefb_gdr.yaml GDASApp. The icefb_GDR.yaml was transferred from godas and is similar aside from the QC.

I am talking about the icefb_GDR.yaml.  It needs to renamed with "gdr" and the anchors inside the yaml as well.

File: parm/soca/obs/config/icefb_GDR.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+obs space:
```
Comment (guillaumevernieres) [inline parm/soca/obs/config/icefb_GDR.yaml]: As pointed out by @ShastriPaturi , that file needs to be renamed `icefb_GDR.yaml` ... Not your bug @AndrewEichmann-NOAA , but we might as well take cor of it now.

Comment (AndrewEichmann-NOAA) [reply parm/soca/obs/config/icefb_GDR.yaml]: You mean renamed from `icefb_GDR.yaml` to `icefb_gdr.yaml`, replacing the existing` icefb_gdr.yaml`?

Comment (AndrewEichmann-NOAA) [reply parm/soca/obs/config/icefb_GDR.yaml]: done

**Review by guillaumevernieres — APPROVED** [review]
:tada:

**Pull request merged**